presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst

```diff
@@ -0,0 +1,97 @@
+             Output partitioning: SINGLE []                                                                                                                           >
+             Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                                            >
+             - Project[PlanNodeId 392][projectLocality = LOCAL] => [orderpriority:varchar(15), count:bigint]                                                          >
+                     Estimates: {source: HistoryBasedSourceInfo, rows: 5 (117B), cpu: ?, memory: ?, network: ?}                                                       >
+                 - Aggregate(FINAL)[orderpriority][$hashvalue][PlanNodeId 4] => [orderpriority:varchar(15), $hashvalue:bigint, count:bigint]                          >
+                         Estimates: {source: HistoryBasedSourceInfo, rows: 5 (117B), cpu: ?, memory: ?, network: ?}                                                   >
+                         count := "presto.default.count"((count_8)) (1:50)                                                                                            >
+                     - LocalExchange[PlanNodeId 354][HASH][$hashvalue] (orderpriority) => [orderpriority:varchar(15), count_8:bigint, $hashvalue:bigint]              >
+                         - RemoteSource[2] => [orderpriority:varchar(15), count_8:bigint, $hashvalue_9:bigint]
+
+    
+Optimizations using HBO
+-----------------------
+
+DetermineJoinDistributionType & DetermineSemiJoinDistributionType
+-----------------------------------------------------------------
+These two optimizations decide whether to do broadcast or repartition for a join. They use the size of the probe and build input in optimization.
+* Data size recorded from history queries will be used when HBO is enabled.
+* Statistics from cost models are used when HBO statistics are not available or HBO is disabled.
+
+ReorderJoins
+------------
+This optimization reorders the join order based on the size of input and output. Data size recorded from history queries will be used when HBO is enabled.
+
+PushPartialAggregationThroughExchange
+-------------------------------------
+This optimization decides whether to split an aggregation into partial and final aggregations. HBO has special session properties to control it.
+* Set `track_partial_aggregation_history` to true to track the output size of the partial aggregation node.
+* Set `use_partial_aggregation_history` to true to use the partial aggregation node statistics to decide whether to split aggregation. The track of partial aggregation statistics
+targets the pattern we found in production queries, where the final aggregation node is cardinality reducing but partial aggregation is not.
+When `use_partial_aggregation_history` is not enabled or partial aggregation statistics are not available, it will fall back to using the final aggregation statistics.
```

Comment:
```suggestion
When `use_partial_aggregation_history` is not enabled or partial aggregation statistics are not available, it will fall back to using the final aggregation statistics.

```
The screenshot in my suggestion for line 77 shows the absence of the edit suggested here.